src/components/text-input/index.md.njk

```diff
@@ -21,7 +21,7 @@ Do not use the text input component if you need to let users enter longer answer
 
 ## How it works
 
-All text inputs must have visible labels; placeholder text is not an acceptable replacement for a label as it vanishes when users start typing.
+All text inputs must have visible labels. Do not use placeholder text instead of a label, as the text vanishes when users start typing. You should also not use placeholder text in hints or examples. Placeholder text is low contrast, and not all screen readers support it.
```

Comment:
For the line 'You should also not use placeholder text in hints or examples.' I think this might need a bit of a rework.

Instead of 'in hints' should it instead be 'for hints' or 'in place of hints'?

```suggestion
All text inputs must have visible labels. Do not use placeholder text in place of a label, as the text vanishes when users start typing. You should also not use placeholder text for hints or examples. Browser default styles for placeholder text are often low contrast and not all screen readers read out placeholder text.
```

Reply:
I also found this reference from W3C talking about how screen readers treat placeholder text, in case we need a reference for that statement:
https://www.w3.org/WAI/tutorials/forms/instructions/#:~:text=Assistive%20technologies%2C%20such%20as%20screen,displayed%20in%20older%20web%20browsers.

Reply:
The majority of the time we want visible labels, but not *all*.  

I think it's acceptable to *not* have a visible label when you're progressively revealing a text input with radios or checkboxes and the input would otherwise be identical to the parent label.

I had this before and the consensus of the #accessibility community was that making the label visually hidden was the clearer thing to do.

A not great example from a previous service:
<img width="798" alt="Screenshot 2022-04-25 at 12 30 42" src="https://user-images.githubusercontent.com/2204224/165080648-c8aeae76-c78e-4034-a16c-ac86598e6b85.png">

Sometimes you can word the two differently - but I've had cases where you're either repeating the content, making the page look awkward. Or else you repeat the parent legend. In those cases, visually hiding the label seemed clearer than showing it.

Reply:
@davidc-gds I'd be very fine with not including an example - my main concern was the very authoritative "All text inputs must have visible labels." which leaves no room for doubt or discussion. Some teams take the design system guidance very literally.

To play devil's advocate - do you need the first sentence at all? Or does it need to mention `visible`?

Reply:
Hey @edwardhorsford in the latest commit we have it written to separate out the 'have a label' and 'have the label visible' statements, which I believe addresses your concern about the authoritative 'must be visible' piece.

`All text inputs must have labels, and in most cases the label should be visible.`

Does that clear up the piece where devil's advocate is coming into play, or does that still apply to this new commit we have, shown above?

Reply:
@davidc-gds It does, though I share the concern about this edge case being too prominent. Could the paragraph just say `All text inputs must have labels` and say nothing more?

Reply:
Hi @edwardhorsford! Currently the paragraph reads:

> All text inputs must have labels, and in most cases the label should be visible.

Is that ok?

Reply:
@EoinShaughnessy yes good.

Reply:
Cheers Ed!